peer.go

```diff
@@ -441,6 +440,7 @@ func newChanMsgStream(f *fundingManager, h *htlcswitch.Switch, p *peer,
 
 // Start starts the chanMsgStream.
 func (c *chanMsgStream) Start() {
+	c.wg.Add(1)
```

Comment:
IMO, the preferred pattern, specifically for private go routines, is:
```
thing.wg.Add(1)
go thing.doSomething()
```
and that this should _imply_ that `thing` will (or at least should) always defer or signal Done to the wait group when that function exits.  Overall I think this improves readability, since we don't incur the extra lines from anonymous functions, and having a consistent pattern makes it easy for others to figure out which wait group is watching, and where it will be released.  It also helps ensure that each class is fully responsible for the go routines it spawns, as they are a resource that must be managed, just like any other.  Then, if you see a go routine spawned without incrementing a wait group, the mental alarms should go off.  Fortunately then, we can all apply this simple pattern to resolve such cases---unless there is a _very_ good reason to not do be watching a particular go routine.  I think it would make more sense to use the first pattern in the event that `msgConsumer` was designed for both synchronous and asynchronous invocation, and did not necessarily signal Done on its own. But in this case, we only use it as a go routine :)